src/test/App.test.js

```diff
@@ -0,0 +1,227 @@
+import {
+  render, screen, act, fireEvent,
+} from '@testing-library/react';
+import axios from 'axios';
+import { Provider } from 'react-redux';
+import {
+  MemoryRouter, Route, Routes,
+} from 'react-router-dom';
+import { API_NOBEL } from '../redux/nationSlice/NationSlice';
+import store from '../redux/store';
+import App from '../App';
+import Allroutes from '../routes/Allroutes';
+import CityRoute from '../routes/CityRoute';
+import NationRoute from '../routes/NationRoute';
+import Winner from '../routes/Winner';
+import Nation from '../components/Nation';
+import NationItem from '../components/NationItem';
+import NationRouteItem from '../routes/NationRouteItem';
+import MyHeader from '../components/Header';
+
+jest.mock('axios');
+
```

Comment:
- [OPTIONAL] You did great writing `unit and integration tests` for all your major components and functions, but following the best practices, I'll suggest you break down these tests into individual multiple files for each component. it will make the test file less bulky and easy to maintain. :+1: :+1: